js/service/signatureservice.js

```diff
@@ -58,15 +58,15 @@ define(function(require) {
 	}
 
 	/**
-	 * @param accountId
+	 * @param account
 	 * @param signature
 	 * @returns {undefined}
 	 */
-	function saveAccountSignature(accountId, signature) {
+	function saveAccountSignature(account, signature) {
 		var defer = $.Deferred();
 
 		var url = OC.generateUrl('/apps/mail/accounts/{id}/signature', {
-			id: accountId
+			id: account.accountId
```

Comment:
did you test this? I assume ``account.accountId`` will be undefined as it should be a Backbone model and you have to use its `get` method instead to access the id.